Deployment health alerts (#3984)
astronomer/docs: astro/alerts.md

:::

#### Deployment health triggers

<HostedBadge/>

> :::privatepreview

Reviewer: Do we have a precedent for linking a contact email in our `Private Preview` banner? Would love to use the docs as a way to generate interest in the feature and get customers to sign up.

Author: We do not and I haven't had great luck with that strategy in general for docs.

:::

> In addition to trigger types that cause alerts for DAGs, you can also create alerts with triggers that send alerts for [Deployment health incidents](deployment-health-incidents.md#deployment-incidents). You can use these alerts for proactive notification when Deployment health issues arise. For example, when the Airflow metadata database storage is unusually high. These alerts:

Reviewer 2: ```suggestion
In addition to creating trigger types that cause alerts for DAGs, you can also create alerts with triggers that send alerts for [Deployment health incidents](deployment-health-incidents.md#deployment-incidents). You can use these alerts for proactive notification when Deployment health issues arise. For example, you can create an alert for when the Airflow metadata database storage is unusually high. These alerts:
```

> In addition to creating trigger types that cause alerts for DAGs, you can also create alerts with triggers that send alerts for [Deployment health incidents](deployment-health-incidents.md#deployment-incidents). You can use these alerts for proactive notification when Deployment health issues arise. For example, you can create an alert for when the Airflow metadata database storage is unusually high. These alerts:

Reviewer: Nit: `you can also create alerts with triggers that send alerts` is a bit redundant, perhaps `you can also create triggers that send alerts` instead?

> In addition to creating trigger types that cause alerts for DAGs, you can also create alerts with triggers that send alerts for [Deployment health incidents](deployment-health-incidents.md#deployment-incidents). You can use these alerts for proactive notification when Deployment health issues arise. For example, you can create an alert for when the Airflow metadata database storage is unusually high. These alerts:

Author: ```suggestion
In addition to creating trigger types that cause alerts for DAGs, you can also create alerts with triggers that notify you about [Deployment health incidents](deployment-health-incidents.md#deployment-incidents). You can use these alerts for proactive notification when Deployment health issues arise. For example, you can create an alert for when the Airflow metadata database storage is unusually high. These alerts:
```

> - Use Astro architecture to identify infrastructure-level incidents that are otherwise hard to monitor

Reviewer 2: ```suggestion
- Use Astro architecture to identify infrastructure-level incidents that are otherwise hard to monitor.
```

> - Use Astro architecture to identify infrastructure-level incidents that are otherwise hard to monitor.

Reviewer: similar comment as what I left on release notes, whatever we decide on should be consistent https://github.com/astronomer/docs/pull/3983/files#r1695985788

- Give you granular information about your Deployment's performance, with more detail than **Healthy** or **Unhealthy**.

> - Incorporate Astro's suggestions about Deployment health into your existing alerting workflows, including email, Slack, Pagerduty, to allow you to respond faster.

Reviewer 2: ```suggestion
- Incorporate Astro's suggestions about Deployment health into your existing alerting workflows, including email, Slack, and Pagerduty, to allow you to respond faster.
```

> - Use Astro architecture to identify infrastructure-level incidents that are otherwise hard to monitor.
> - Give you granular information about your Deployment's performance, with more detail than **Healthy** or **Unhealthy**.
> - Incorporate Astro's suggestions about Deployment health into your existing alerting workflows, including email, Slack, and Pagerduty, to allow you to respond faster.

Author: ```suggestion
Using Deployment health alerts, you can:
- Improve alerting coverage beyond DAG and task failures to address infrastructure-level incidents that are otherwise difficult to monitor.
- Proactively monitor Deployment health and take immediate remediation actions through email, Slack, and PagerDuty to reduce average time to resolution.
- Share Deployment health visibility across teams.
```

> The following alerts can be set to send you a notification for specific health incidents:

Reviewer 2: ```suggestion
You can set the following alerts to send you a notification for specific health incidents:
```

:::

#### Deployment health triggers

<HostedBadge/>

:::privatepreview
:::

> In addition to creating trigger types that cause alerts for DAGs, you can also create alerts with triggers that notify you about [Deployment health incidents](deployment-health-incidents.md#deployment-incidents). You can use these alerts for proactive notification when Deployment health issues arise. For example, you can create an alert for when the Airflow metadata database storage is unusually high. These alerts:

Reviewer: "These alerts" should be deleted

> In addition to creating trigger types that cause alerts for DAGs, you can also create alerts with triggers that notify you about [Deployment health incidents](deployment-health-incidents.md#deployment-incidents). You can use these alerts for proactive notification when Deployment health issues arise. For example, you can create an alert for when the Airflow metadata database storage is unusually high. These alerts:

Author: ```suggestion
In addition to creating trigger types that cause alerts for DAGs, you can also create alerts with triggers that notify you about [Deployment health incidents](deployment-health-incidents.md#deployment-incidents). You can use these alerts for proactive notification when Deployment health issues arise. For example, you can create an alert for when the Airflow metadata database storage is unusually high.
```

Using Deployment health alerts, you can:
- Improve alerting coverage beyond DAG and task failures to address infrastructure-level incidents that are otherwise difficult to monitor.

> - Proactively monitor Deployment health and take immediate remediation actions through email, Slack, and PagerDuty to reduce average time to resolution.

Reviewer: I specifically wanted to include "mean time to resolution" as it's an [incident response metric](https://www.appdynamics.com/topics/what-is-mttr) that customers care about (and that actionable alerts can help reduce). I was thinking about this in terms of MTTR, which actually can mean a few things (mean time to resolve, repair, respond). I'd be comfortable including the acronym itself `MTTR (mean time to resolution)` if that is clearer!

> Using Deployment health alerts, you can:
> - Improve alerting coverage beyond DAG and task failures to address infrastructure-level incidents that are otherwise difficult to monitor.
> - Proactively monitor Deployment health and take immediate remediation actions through email, Slack, and PagerDuty to reduce average time to resolution.

Author: ```suggestion
Using Deployment health alerts, you can:
- Improve alerting coverage beyond DAG and task failures to address infrastructure-level incidents that are otherwise difficult to monitor.
- Proactively monitor Deployment health and take immediate remediation actions through email, Slack, and PagerDuty to reduce mean time to resolution.
```
